src/server/bg_services/log_replication_scanner.js

```diff
@@ -0,0 +1,206 @@
+/* Copyright (C) 2016 NooBaa */
+'use strict';
+
+const _ = require('lodash');
+const system_store = require('../system_services/system_store').get_instance();
+const dbg = require('../../util/debug_module')(__filename);
+const system_utils = require('../utils/system_utils');
+const config = require('../../../config');
+const P = require('../../util/promise');
+const Semaphore = require('../../util/semaphore');
+const replication_store = require('../system_services/replication_store').instance();
+const cloud_utils = require('../../util/cloud_utils');
+const log_parser = require('./replication_log_parser');
+const replication_utils = require('../utils/replication_utils');
+
+// eslint-disable-next-line no-unused-vars
+class LogReplicationScanner {
+
+    /**
+     * @param {{
+     *   name: string;
+     *   client: nb.APIClient;
+     * }} params
+     */
+    constructor({ name, client }) {
+        this.name = name;
+        this.client = client;
+        this._scanner_sem = new Semaphore(config.REPLICATION_SEMAPHORE_CAP, {
+            timeout: config.REPLICATION_SEMAPHORE_TIMEOUT,
+            timeout_error_code: 'LOG_REPLICATION_ITEM_TIMEOUT',
+            verbose: true
+        });
+        this.noobaa_connection = undefined;
+    }
+
+    async run_batch() {
+        if (!this._can_run()) return;
+        dbg.log0('log_replication_scanner: starting scanning bucket replications');
+        try {
+            if (!this.noobaa_connection) {
+                this.noobaa_connection = cloud_utils.set_noobaa_s3_connection(system_store.data.systems[0]);
+            }
+            await this.scan();
+        } catch (err) {
+            dbg.error('log_replication_scanner:', err, err.stack);
```

Comment:
When you print `err` does it also prints the stack (`err.stack`)?
If so then you print the error stack twice.

Reply:
Yeah, it seem redundant. Will remove it.

Reply:
Marking resolved - addressed.